### PR TITLE
Worldmap

### PR DESCRIPTION
**Pull request by OrN**
Adds a world map that can be accessed via clicking Map icon, or pressing alt + m. Also introduces some changes to login message and settings having a button on the wrench.

The map can do a lot of things, waypoints, searching, object searching, hightlighting, tracking player position.

Search feature will return the closest search to the player if you are logged in.

Needs some testing.

Untitled

### CODE REVIEW

File: src/Client/WorldMapWindow.java [hunk 1]
```diff
@@ -0,0 +1,1693 @@
...
+                setAlpha(g, 1.0f);
+            }
+        }
+
+        g.dispose();
+        mapImageUpdate[plane] = true;
+    }
+
+    private static void updateMapRender() {
+        System.out.println("update");
```
Comment (Hubcapp) [inline src/Client/WorldMapWindow.java:332]: oops

File: src/Client/WorldMapWindow.java [hunk 2]
```diff
@@ -0,0 +1,1693 @@
...
+                        public void run() {
+                            runInit();
+                        }
+                    });
+        } catch (InvocationTargetException e) {
+            Logger.Error("There was a thread-related error while setting up the replay queue window!");
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            Logger.Error(
+                    "There was a thread-related error while setting up the replay queue window! The window may not be initialized properly!");
```
Comment (Hubcapp) [inline src/Client/WorldMapWindow.java:190]: this is not the replay queue window

File: src/Client/WorldMapWindow.java [hunk 3]
```diff
@@ -0,0 +1,1693 @@
...
+        try {
+            SwingUtilities.invokeAndWait(
+                    new Runnable() {
+                        @Override
+                        public void run() {
+                            runInit();
+                        }
+                    });
+        } catch (InvocationTargetException e) {
+            Logger.Error("There was a thread-related error while setting up the replay queue window!");
```
Comment (Hubcapp) [inline src/Client/WorldMapWindow.java:186]: not the replay queue window

**Review by Hubcapp — CHANGES_REQUESTED** [review]

**Comment by Hubcapp** [comment]
```
[  INFO][20:16:07] Loading resource: /assets/map/scenery.bin
java.lang.IllegalArgumentException: URI is not hierarchical
	at java.io.File.<init>(File.java:418)
	at Client.WorldMapWindow.initScenery(Unknown Source)
	at m.a(Unknown Source)
	at client.f(Unknown Source)
	at client.a(Unknown Source)
	at e.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by Hubcapp** [comment]
Would be nice to support CTRL-A to highlight all text in the "Type to search..." box, maybe for future development.

**Comment by Hubcapp** [comment]
LGTM, pending feedback from @lmsv-mx123

**Comment by Hubcapp** [comment]
FUTURE TODO:
Item spawns on map
"Boundaries" aka "Wall Objects"
5x5 tiles instead of 3x3
Click objects to display info in-map
User configurable colour for highlighted search result, for colourblindness and stuff
specify labels for other floors potentially
POI on other floors
click on things to show info, including the tile
NPC heatmap

**Review by Hubcapp — APPROVED** [review]

**Comment by lmsv-mx123** [comment]
Looks good to me

**Review by lmsv-mx123 — APPROVED** [review]
Sweet, great job

**Pull request merged**